src/spatialdata/_core/spatialdata.py

```diff
@@ -263,115 +256,30 @@ def aggregate(
-
-    def _add_labels_in_memory(
-        self, name: str, labels: SpatialImage | MultiscaleSpatialImage, overwrite: bool = False
-    ) -> None:
-        """
-        Add a labels element to the SpatialData object.
-
-        Parameters
-        ----------
-        name
-            name of the labels
-        labels
-            the labels element to be added
-        overwrite
-            whether to overwrite the labels if the name already exists.
-        """
-        self._validate_unique_element_names(
-            list(self.images.keys()) + list(self.points.keys()) + list(self.shapes.keys()) + [name]
-        )
-        if name in self._labels and not overwrite:
-            raise KeyError(f"Labels {name} already exists in the dataset.")
-        ndim = len(get_axes_names(labels))
-        if ndim == 2:
-            Label2D_s.validate(labels)
-            self._labels[name] = labels
-        elif ndim == 3:
-            Label3D_s.validate(labels)
-            self._labels[name] = labels
-        else:
-            raise ValueError(f"Only yx and zyx labels supported, got {ndim} dimensions")
-
-    def _add_shapes_in_memory(self, name: str, shapes: GeoDataFrame, overwrite: bool = False) -> None:
-        """
-        Add a shapes element to the SpatialData object.
-
-        Parameters
-        ----------
-        name
-            name of the shapes
-        shapes
-            the shapes element to be added
-        overwrite
-            whether to overwrite the shapes if the name already exists.
-        """
-        self._validate_unique_element_names(
-            list(self.images.keys()) + list(self.points.keys()) + list(self.labels.keys()) + [name]
-        )
-        if name in self._shapes and not overwrite:
-            raise KeyError(f"Shapes {name} already exists in the dataset.")
-        Shape_s.validate(shapes)
-        self._shapes[name] = shapes
-
-    def _add_points_in_memory(self, name: str, points: DaskDataFrame, overwrite: bool = False) -> None:
-        """
-        Add a points element to the SpatialData object.
-
-        Parameters
-        ----------
-        name
-            name of the points element
-        points
-            the points to be added
-        overwrite
-            whether to overwrite the points if the name already exists.
-        """
-        self._validate_unique_element_names(
-            list(self.images.keys()) + list(self.labels.keys()) + list(self.shapes.keys()) + [name]
-        )
-        if name in self._points and not overwrite:
-            raise KeyError(f"Points {name} already exists in the dataset.")
-        Point_s.validate(points)
-        self._points[name] = points
-
     def is_backed(self) -> bool:
         """Check if the data is backed by a Zarr storage or if it is in-memory."""
         return self.path is not None
 
+    @property
+    def path(self) -> Path | None:
+        """Path to the Zarr storage."""
+        return self._path
+
+    @path.setter
```

Comment:
I have a comment on this setter, not sure if we should disallow it. See GitHub discussion. (I added a similar "See GitHub discussion" comment in other places, they all refer to the same discussion.)

Reply:
Yes, I think this setter should not be public.